tests/test_sleep_windows.py

```diff
@@ -0,0 +1,48 @@
+import datetime
+import unittest
+
+from greenpithumb import sleep_windows
+
+
+class TestSleepWindowParser(unittest.TestCase):
+
+    def test_parser_valid_inputs(self):
+        self.assertEqual(
+            sleep_windows.parse(['01:00-03:15']),
+            [(datetime.time(1, 0), datetime.time(3, 15))])
+        self.assertEqual(
+            sleep_windows.parse(['01:00-03:15', '13:45-16:00']),
+            [(datetime.time(1, 0), datetime.time(3, 15)),
+             (datetime.time(13, 45), datetime.time(16, 0))])
+        # Overlapping ranges OK.
+        self.assertEqual(
+            sleep_windows.parse(['01:00-03:15', '02:45-04:00']),
+            [(datetime.time(1, 0), datetime.time(3, 15)),
+             (datetime.time(2, 45), datetime.time(4, 0))])
+        # Subset ranges OK.
+        self.assertEqual(
+            sleep_windows.parse(['01:00-03:15', '02:45-03:00']),
+            [(datetime.time(1, 0), datetime.time(3, 15)),
+             (datetime.time(2, 45), datetime.time(3, 0))])
+
+    def test_parser_invalid_inputs(self):
+        invalid_inputs = [
+            [''],  # empty string
+            [10],  # not a string
+            ['MickeyGates'],  # gibberish
+            ['01:00-0300'],  # missing colon
+            ['01:00-'],  # incomplete
+            ['100:50-03:00'],  # invalid hour
+            ['10:500-12:00'],  # invalid minute
+            ['05:00:09.897-06:00']  # unsupported seconds-level precision
+        ]
+        for invalid_input in invalid_inputs:
+            with self.assertRaises(sleep_windows.InvalidWindowFormatError):
+                sleep_windows.parse(invalid_input)
+
+    def test_hypersomnia(self):
+        hypersomnia_inputs = [['00:00-00:00'], ['01:00-12:00', '12:00-01:00'],
```

Comment:
(minor) Suggest using parens for the outer set just to make the distinction a little clearer like

```python
hypersomnia_inputs = (['00:00-00:00'], ...
```